From 924bff95512505a13b19a2d2865d2fe4df2444e1 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Mon, 18 Mar 2024 08:19:25 +0000
Subject: [PATCH 13/16] ice: remove unused netdev_unregistering()

Accommodate the changes made in commit f3fb589aeb88 ("net: remove
unused netdev_unregistering()"), which was merged in the v6.4
development cycle, and which removed the function
netdev_unregistering(). This causes build errors to be encountered,
which are listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?id=f3fb589aeb88

Resolved compilation errors:

src/ice_lag.c:280:7: error: implicit declaration of function netdev_unregistering;\
 did you mean devl_unregister [-Werror=implicit-function-declaration]
 280 |       netdev_unregistering(lag_work->lag->upper_netdev))

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/ice_lag.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ice_lag.c b/src/ice_lag.c
index 3722763..717e625 100644
--- a/src/ice_lag.c
+++ b/src/ice_lag.c
@@ -277,7 +277,7 @@ static void ice_lag_process_event(struct work_struct *work)
 	case NETDEV_UNREGISTER:
 		netdev = lag_work->info.bonding_info.info.dev;
 		if (netdev == lag_work->lag->netdev && lag_work->lag->bonded &&
-		    netdev_unregistering(lag_work->lag->upper_netdev))
+		    lag_work->lag->upper_netdev->reg_state == NETREG_UNREGISTERING)
 			ice_lag_unlink(lag_work->lag);
 		break;
 	default:
-- 
2.43.0

